osu.Game.Rulesets.Osu/Difficulty/Evaluators/ReadingEvaluator.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                OsuDifficultyHitObject hitObject = (OsuDifficultyHitObject)current.Next(i);
+
+                if (hitObject.IsNull() ||
+                    hitObject.StartTime - current.StartTime > reading_window_size ||
+                    current.StartTime + preempt < hitObject.StartTime) // Object not visible at the time current object needs to be clicked.
+                    break;
+
+                double timeBetweenCurrAndLoopObj = hitObject.StartTime - current.StartTime;
+                double timeNerfFactor = getTimeNerfFactor(timeBetweenCurrAndLoopObj);
+
+                visibleObjectCount += hitObject.OpacityAt(current.BaseObject.StartTime, false) * timeNerfFactor;
+            }
+
+            return visibleObjectCount;
+        }
+
+        // Returns the amount of time a note spends invisible with the hidden mod at the current approach rate.
+        private static double getDurationSpentInvisible(OsuDifficultyHitObject current)
+        {
+            var baseObject = (OsuHitObject)current.BaseObject;
+
+            double fadeOutStartTime = baseObject.StartTime - baseObject.TimePreempt + baseObject.TimeFadeIn;
+            double fadeOutDuration = baseObject.TimePreempt * OsuModHidden.FADE_OUT_DURATION_MULTIPLIER;
+
+            return (fadeOutStartTime + fadeOutDuration) - (baseObject.StartTime - baseObject.TimePreempt);
+        }
+
+        // Returns a factor of how often the current object's angle has been repeated in a certain time frame.
+        // It does this by checking the difference in angle between current and past objects and sums them based on a range of similarity.
+        // https://www.desmos.com/calculator/eb057a4822
+        private static double getConstantAngleNerfFactor(OsuDifficultyHitObject current)
+        {
+            const double time_limit = 2000; // 2 seconds
+            const double time_limit_low = 200;
+
+            double constantAngleCount = 0;
+            int index = 0;
+            double currentTimeGap = 0;
+
+            while (currentTimeGap < time_limit)
+            {
+                var loopObj = (OsuDifficultyHitObject)current.Previous(index);
+
+                if (loopObj.IsNull())
+                    break;
+
+                // Account less for objects that are close to the time limit.
+                double longIntervalFactor = Math.Clamp(1 - (loopObj.AdjustedDeltaTime - time_limit_low) / (time_limit - time_limit_low), 0, 1);
+
+                if (loopObj.Angle.IsNotNull() && current.Angle.IsNotNull())
+                {
+                    double angleDifference = Math.Abs(current.Angle.Value - loopObj.Angle.Value);
+                    double stackFactor = DifficultyCalculationUtils.Smootherstep(loopObj.LazyJumpDistance, 0, OsuDifficultyHitObject.NORMALISED_RADIUS);
+
+                    constantAngleCount += Math.Cos(3 * Math.Min(Math.PI / 6, angleDifference * stackFactor)) * longIntervalFactor;
```

Comment:
```suggestion
                    constantAngleCount += Math.Cos(3 * Math.Min(double.DegreesToRadians(30), angleDifference * stackFactor)) * longIntervalFactor;
```

For consistency with all the other angle-based calculations